src/call.cpp

```diff
@@ -920,36 +920,69 @@ char * call::get_last_header(const char * name)
         ERROR("call::get_last_header: Header to parse bigger than %d (%zu)", MAX_HEADER_LEN, strlen(name));
     }
 
+    char header_name[MAX_HEADER_LEN];
+    const char *sep = strrchr(name, ':');
+
+    if (!sep){ /* [last_Header] */
+        sprintf(header_name, "%s:", name);
+        return get_header(last_recv_msg, header_name, false);
+    }
+
+    snprintf(header_name, len - strlen(sep) + 2, name);
+
+    /* [last_Header:] */
     if (name[len - 1] == ':') {
         return get_header(last_recv_msg, name, false);
+    }
+    char *value = get_header(last_recv_msg, header_name, true);
```

Comment:
Apparently you're not using value `get_header(last_recv_msg, header_name, true);` here, move it to the `; /* done */` instead.

It would be only useful here if we used functions that operate on that single header directly later on, e.g.:
* `get_param(value, "tag");` or `extract_param(value, "tag");`
* `get_first_uri(value);` or `extract_first_uri(value);`

They could certainly be implemented and would clean up things some more.